app/models/admin_user.rb

```diff
@@ -23,11 +27,54 @@ class AdminUser < ActiveRecord::Base
   validates_format_of :password, with: PASSWORD_REGEX, allow_blank: true
   validates_inclusion_of :role, in: ROLES
 
+  # = Callbacks =
+  before_update if: :crypted_password_changed? do
+    self.force_password_reset = false
+    self.password_changed_at = Time.current
+  end
+
   # = Finders =
   scope :by_name, -> { order(:last_name, :first_name) }
   scope :by_role, ->(role) { where(role: role).order(:id) }
 
   # = Methods =
+  def current_password
+    defined?(@current_password) ? @current_password : nil
+  end
+
+  def current_password=(value)
```

Comment:
Could we replace this method (and the one above) with `attr_accessor :current_password` ?